samples/net/sockets/big_http_download/Kconfig

```diff
@@ -11,4 +11,10 @@ config SAMPLE_BIG_HTTP_DL_URL
 	help
 	  URL to download.
 
+config SAMPLE_BIG_HTTP_DL_NUM_ITER
+	int "Limit number of iterations"
+	default -1
```

Comment:
0 would feel more intuitive as a unlimited value, as value of 0 would not make sense here. Same applies of course also to <0 values which are not meaningful here. That would require small changes to while loop, perhaps easiest is to change it to `do {} while();`

Reply:
> 0 would feel more intuitive as a unlimited value, as value of 0 would not make sense here.

Well, 0 means "zero iterations". Whether that makes sense to use is up to users. I also wouldn't agree that it's more intuitive, because 0 is again, zero times, and it follows convention of POSIX select()/poll() calls, which take 0, or positive number, or negative (commonly -1) as infinite delay.

Are you sure this request is not nit-picking? If you'd insist that value of 0 is used in Kconfig, then my fix on the source side would be to add:

~~~
if (num_iter == 0) {
    num_iter = -1;
}
~~~

Reply:
0 zero value here would not make sense as you would not do any iterations then, so it is a special value. 
If using value 0 as limitless, then just do
```
do {

} while (--num_iterations);
```
below.

Reply:
> } while (--num_iterations);

Ok, updated as suggested.